tedana/workflows/tedana_reclassify.py

```diff
@@ -0,0 +1,484 @@
+    # First, make the output generator
+    comp_names = comptable["Component"].values
+    mixing_df = pd.DataFrame(data=mmix, columns=comp_names)
+    to_copy = [
+        "z-scored ICA components img",
+        "ICA mixing tsv",
+        "ICA decomposition json",
+        "ICA metrics json",
+    ]
+    if used_gs:
+        to_copy.append(gskey)
+        to_copy.append("has gs combined img")
+
+    for tc in to_copy:
+        print(tc)
+        io_generator.save_file(ioh.get_file_contents(tc), tc)
+
+    # Save component selector and tree
+    selector.to_files(io_generator)
+
+    if selector.n_accepted_comps == 0:
+        LGR.warning(
+            "No accepted components remaining after manual classification! "
+            "Please check data and results!"
+        )
+
+    mmix_orig = mmix.copy()
+    # TODO: make this a function
+    if tedort:
+        comps_accepted = selector.accepted_comps
+        comps_rejected = selector.rejected_comps
+        acc_ts = mmix[:, comps_accepted]
+        rej_ts = mmix[:, comps_rejected]
+        betas = np.linalg.lstsq(acc_ts, rej_ts, rcond=None)[0]
+        pred_rej_ts = np.dot(acc_ts, betas)
+        resid = rej_ts - pred_rej_ts
+        mmix[:, comps_rejected] = resid
+        comp_names = [
+            io.add_decomp_prefix(comp, prefix="ica", max_value=comptable.index.max())
+            for comp in range(selector.n_comps)
+        ]
+        mixing_df = pd.DataFrame(data=mmix, columns=comp_names)
+        io_generator.save_file(mixing_df, "ICA orthogonalized mixing tsv")
+        RepLGR.info(
+            "Rejected components' time series were then "
+            "orthogonalized with respect to accepted components' time "
+            "series."
+        )
+
+    n_vols = data_oc.shape[3]
+    img_t_r = io_generator.reference_img.header.get_zooms()[-1]
+    adaptive_mask = utils.reshape_niimg(adaptive_mask)
+    mask_denoise = adaptive_mask >= 1
+    data_oc = utils.reshape_niimg(data_oc)
+
+    # TODO: make a better result-writing function
+    # #############################################!!!!
+    # TODO: make a better time series creation function
+    #       - get_ts_fit_tag(include=[], exclude=[])
+    #       - get_ts_regress/residual_tag(include=[], exclude=[])
+    #       How to handle [acc/rej] + tag ?
+    io.writeresults(
+        data_oc,
+        mask=mask_denoise,
+        comptable=comptable,
+        mmix=mmix,
+        n_vols=n_vols,
+        io_generator=io_generator,
+    )
+
+    if mir:
+        io_generator.force = True
+        gsc.minimum_image_regression(data_oc, mmix, mask_denoise, comptable, io_generator)
+        io_generator.force = False
+
+    # Write out BIDS-compatible description file
+    derivative_metadata = {
+        "Name": "tedana Outputs",
+        "BIDSVersion": "1.5.0",
+        "DatasetType": "derivative",
+        "GeneratedBy": [
+            {
+                "Name": "tedana_reclassify",
+                "Version": __version__,
```

Comment:
Not sure what you're asking here. Do you want to store the version of `tedana_reclassify` and also maintain the version for the original `tedana` that was run?

Reply:
I was thinking that the reclassification workflow could compare its own version against the tedana version listed in the incoming dataset's dataset_description, and raise a warning if the versions are different, since major version differences would mean that the incoming data and workflow being run might be incompatible (e.g., if filenames were changed). We are going to try to ensure that doesn't happen, so I think a warning makes more sense than an error.